### ISSUE TITLE
usize overflow in Vec::reserve

### ISSUE THREAD

**Issue by codahale** [hyperium/hyper#412]
* hyper 0.3.6
* rustc 1.0.0-nightly (6cf3b0b74 2015-03-30) (built 2015-03-31)
* Linux 3.14.35-28.38.amzn1.x86_64 #1 SMP Wed Mar 11 22:50:37 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
* wrk @ a209691

I’m using the hello-world example from the `examples` dir. This only happens with release builds, and it only happens after a number of successful requests (~12,000+):

```
thread '<unnamed>' panicked at 'Vec::reserve: `usize` overflow', /home/rustbuild/src/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libcore/option.rs:332
thread '<unnamed>' panicked at 'child thread None panicked', /home/rustbuild/src/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/thread/mod.rs:797
thread '<main>' panicked at 'child thread None panicked', /home/rustbuild/src/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/thread/mod.rs:797
```

**Comment by seanmonstar** [comment]
Could you run again with RUST_BACKTRACE=1?

**Comment by seanmonstar** [comment]
Perhaps fixed in 0.3.7?

**Comment by codahale** [comment]
I can’t repro on 0.3.7, so whatever the problem was it’s solved now.
